packages/backend/convex/http.ts

```diff
@@ -1,170 +1,17 @@
-import type { Doc } from "./_generated/dataModel";
-import { type ActionCtx, httpAction } from "./_generated/server";
+import { api, components, internal } from "./_generated/api";
 import { auth } from "./auth";
-import {
-  sendSubscriptionErrorEmail,
-  sendSubscriptionSuccessEmail,
-} from "./email/templates/subscriptionEmail";
-import { env } from "./env";
-
-const handleUpdateSubscription = async (
-  ctx: ActionCtx,
-  user: Doc<"users">,
-  subscription:
-    | WebhookSubscriptionCreatedPayload
-    | WebhookSubscriptionUpdatedPayload,
-) => {
-  const subscriptionItem = subscription.data;
-  await ctx.runMutation(internal.subscriptions.replaceSubscription, {
-    userId: user._id,
-    subscriptionPolarId: subscription.data.id,
-    input: {
-      productId: subscriptionItem.productId,
-      priceId: subscriptionItem.priceId,
-      interval: subscriptionItem.recurringInterval,
-      status: subscriptionItem.status,
-      currency: "usd",
-      currentPeriodStart: subscriptionItem.currentPeriodStart.getTime(),
-      currentPeriodEnd: subscriptionItem.currentPeriodEnd?.getTime(),
-      cancelAtPeriodEnd: subscriptionItem.cancelAtPeriodEnd,
-    },
-  });
-};
-
-const handleSubscriptionChange = async (
-  ctx: ActionCtx,
-  event: WebhookSubscriptionCreatedPayload | WebhookSubscriptionUpdatedPayload,
-) => {
-  const user = await ctx.runQuery(internal.subscriptions.getPolarEventUser, {
-    polarId: event.data.userId,
-    email: event.data.user.email,
-  });
-  if (!user?.email) {
-    throw new Error("User not found");
-  }
-
-  await handleUpdateSubscription(ctx, user, event);
-
-  const freePlan = await ctx.runQuery(internal.subscriptions.getPlanByKey, {
-    key: "free",
-  });
-
-  // Only send email for paid plans
-  if (event.data.productId !== freePlan?.polarProductId) {
-    await sendSubscriptionSuccessEmail({
-      email: user.email,
-      subscriptionId: event.data.id,
-    });
-  }
-
-  return new Response(null);
-};
-
-const handlePolarSubscriptionUpdatedError = async (
-  ctx: ActionCtx,
-  event: WebhookSubscriptionCreatedPayload | WebhookSubscriptionUpdatedPayload,
-) => {
-  const subscription = event.data;
-
-  const user = await ctx.runQuery(internal.subscriptions.getPolarEventUser, {
-    polarId: subscription.userId,
-    email: subscription.user.email,
-  });
-  if (!user?.email) throw new Error("User not found");
-
-  const freePlan = await ctx.runQuery(internal.subscriptions.getPlanByKey, {
-    key: "free",
-  });
-
-  // Only send email for paid plans
-  if (event.data.productId !== freePlan?.polarProductId) {
-    await sendSubscriptionErrorEmail({
-      email: user.email,
-      subscriptionId: subscription.id,
-    });
-  }
-  return new Response(null);
-};
+import { polar } from "./subscriptions";
 
 const http = httpRouter();
 
 auth.addHttpRoutes(http);
 
-http.route({
+// biome-ignore lint/suspicious/noExplicitAny: <explanation>
+polar.registerRoutes(http as any, {
```

Comment:
interesting to see this is required, I thought this would be taken care of by making convex a peerDependency but you're already doing that

Reply:
Yeah the error I'm avoiding there is about separate declarations of a private property, but the dependency paths are the same. Probably worth another look.